src/main/java/org/apache/commons/collections4/bloomfilter/BitMapProducer.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.collections4.bloomfilter;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.LongPredicate;
+
+/**
+ * Produces bit map longs for a Bloom filter.
+ *
+ * Each bit map is a little-endian long value representing a block of bits of in a filter.
+ *
+ * <p>The returned array will have length {@code ceil(m / 64)} where {@code m} is the
+ * number of bits in the filter and {@code ceil} is the ceiling function.
+ * Bits 0-63 are in the first long. A value of 1 at a bit position indicates the bit
+ * index is enabled.
+ * </p><p><em>
+ * The default implementations of the {@code makePredicate()} and {@code asBitMapArray} methods
+ * are slow and should be reimplemented in the implementing classes where possible.</em></p>
+ *
+ * @since 4.5
+ */
+@FunctionalInterface
+public interface BitMapProducer {
```

Comment:
Naming conventions between the to and from method in this and IndexProducer do not align:
```
IndexProducer:
// Update to use the ... varargs syntax
fromIntArray(int[] array) -> fromIndexArray(int... array)
int[] toIndexArray()

BitMapProducer:
fromLongArray(long... array) -> fromBitMapArray(long... array)
long[] toBitMapArray();
```
I would also order the methods in this interface to match IndexProducer. This means moving `toBitMapArray()` as the last method.